File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = -1

    stack = []
    stack.append(s[0])
    for i in range(1, len(s)):
        if len(stack) == 0:
            stack.append(s[i])
            continue
        if stack[len(stack)-1] == s[i]:
            stack.pop()
            continue
        else:
            stack.append(s[i])
            continue

        
    print(stack)
    
    if len(stack) == 0:
        answer = 1
    else:
        answer = 0
    
    return answer

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution {
    public static int answer = 0;
 
    public static ArrayDeque<String> Q = new ArrayDeque<>();
    public static ArrayDeque<String> tQ = new ArrayDeque<>();
    
    
    public int solution(String s) {
        
        for(int i =0; i < s.length(); i++){
            Q.add(s.charAt(i) + "");
        }    
        
        tQ.add(Q.removeFirst());
        
        while(!Q.isEmpty()){
            String cur = Q.removeFirst();
            
            if(tQ.isEmpty()){
                tQ.add(cur);
                continue;
            }
            
            // System.out.println(tQ.getLast() + " : " + cur);
            
            if(tQ.getLast().contains(cur)){
                tQ.removeLast();
            }else{
                tQ.add(cur);
            }
        }
        
        if(Q.isEmpty() && tQ.isEmpty()){
            answer = 1;
        }
        
        return answer;
    }
}